src/v/random/random_test.cc

```diff
@@ -34,3 +35,19 @@ BOOST_AUTO_TEST_CASE(alphanum_generator) {
         }
     }
 }
+
+BOOST_AUTO_TEST_CASE(alphanum_max_distinct_generator) {
+    constexpr size_t cardinality = 11;
+    absl::flat_hash_set<ss::sstring> strings;
+    for (auto i = 0; i < 100; i++) {
+        auto s = random_generators::gen_alphanum_max_distinct(cardinality);
+        // ensure no \0 in size
+        for (auto j = 0; j < random_generators::alphanum_max_distinct_strlen;
```

Comment:
does `alphanum_max_distinct_strlen` need to be public just so it can be used here? Can you just iterate through the string checking for \0 ?

Reply:
Sure, we just lose testing the property that there aren't any unexpected null characters, and the ability of callers to know the maximum length.